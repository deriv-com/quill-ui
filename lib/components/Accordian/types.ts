import { ReactNode } from "react";
import { AccordionVariants } from ".";
import { TRegularSizes } from "@types";

export type AccordionProps = {
    id?: string;
    className?: string;
    title?: string;
    subtitle?: string;
    icon?: ReactNode;
    color?: string;
    textSize?: TRegularSizes;
    expanded?: boolean;
    divider?: AccordionDivider;
    disabled?: boolean;
    contentClassname?: string;
    expandedColor?: TExpandedColor;
    content?: () => ReactNode;
    customContent?: () => ReactNode;
    isFlush?: boolean;
    onExpand?: (e: boolean, i: string) => void;
};
type TExpandedColor = "white" | "gray";
export interface AccordionTabProps extends Omit<AccordionProps, "content"> {
    tab?: Tab;
    className?: string;
    variant?: keyof AccordionVariants;
    content: {
        className?: string;
        data: AccordionProps[][];
    };
    multiCollapse?: boolean;
}

export type Tab = {
    align?: TabAlign;
    data: TabData[];
};

export type TabData = {
    id: number;
    title: string;
};

export type TabAlign = "center" | "end";

export type AccordionDivider = "none" | "both" | "bottom";
