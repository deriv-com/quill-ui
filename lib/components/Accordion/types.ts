import { ReactNode } from "react";
import { AccordionVariants } from ".";
import { TRegularSizes } from "@types";

export type AccordionProps = {
    id?: string;
    className?: string;
    title?: string;
    subtitle?: string;
    icon?: ReactNode;
    color?: string;
    textSize?: TRegularSizes;
    expanded?: boolean;
    divider?: AccordionDivider;
    disabled?: boolean;
    contentClassname?: string;
    content?: () => ReactNode;
    customContent?: () => ReactNode;
    onExpand?: (e: boolean, i: string) => void;
};

export interface AccordionTabProps extends Omit<AccordionProps, "content"> {
    className?: string;
    variant?: keyof AccordionVariants;
    content: {
        className?: string;
        data: AccordionProps[][];
    };
    multiCollapse?: boolean;
}

export type AccordionDivider = "none" | "both" | "bottom";
