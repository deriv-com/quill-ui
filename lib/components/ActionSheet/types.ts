import { ComponentPropsWithoutRef } from "react";
import { ExcludeAllNull, TLeftOrRight } from "../../types";

interface OpenType {
    isOpen?: boolean | undefined;
    onOpen?: () => void;
}

interface actionSheetRootCVA {
    position?: TLeftOrRight;
    show?: boolean;
}

interface actionSheetFooterCVA {
    alignment?: "vertical" | "horizontal";
}

export interface RootProps
    extends ComponentPropsWithoutRef<"div">,
        ExcludeAllNull<actionSheetRootCVA>,
        OpenType {
    onClose?: () => void;
    type?: "modal" | "non-modal";
    expandable?: boolean;
}

export type RootPosition = RootProps["position"];

interface ActionType {
    content: string;
    onAction: () => void;
}

export interface FooterProps
    extends ComponentPropsWithoutRef<"div">,
        actionSheetFooterCVA {
    primaryAction?: ActionType;
    secondaryAction?: ActionType;
    shouldCloseOnPrimaryButtonClick?: boolean;
    shouldCloseOnSecondaryButtonClick?: boolean;
}

export type FooterAlignment = FooterProps["alignment"];
