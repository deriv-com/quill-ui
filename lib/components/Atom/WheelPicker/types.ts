import { THorizontalPosition } from "@types";
import { HTMLAttributes, ReactNode } from "react";

export type TWheelTypeSelectItem = {
    value: string | number;
    label?: ReactNode;
};
export interface WheelPickerProps extends HTMLAttributes<HTMLElement> {
    data: TWheelTypeSelectItem[];
    selectedValue: string | number;
    setSelectedValue: (value: string | number) => void;
    handleKeyDown?: (e: React.KeyboardEvent) => void;
    isFocused?: boolean;
    dropDownItemClassName?: string;
    containerClassName?: string;
    listClassName?: string;
    position?: THorizontalPosition | undefined;
    containerHeight?: string;
    disabled?: boolean;
}
export interface WheelPickerContainerProps
    extends Omit<
        WheelPickerProps,
        "data" | "setSelectedValue" | "selectedValue"
    > {
    data: TWheelTypeSelectItem[][];
    inputValues?: (string | number)[];
    setInputValues?: (index: number, value: string | number) => void;
    children?: React.ReactNode;
    close?: () => void;
    setSelectedValue?: (value: string | number) => void;
    containerHeight?: string;
}
export interface TimeWheelPickerContainerProps
    extends Omit<
        WheelPickerProps,
        "data" | "setSelectedValue" | "selectedValue"
    > {
    is12Hour: boolean;
    children?: React.ReactNode;
    minutesInterval?: number;
    hoursInterval?: number;
    locale?: string;
    close?: () => void;
    setSelectedValue?: (value: string | number) => void;
    startTimeIn24Format?: string;
    endTimeIn24Format?: string;
    selectedTime?: string;
    setSelectedTime?: (value: string) => void;
    data?: TWheelTypeSelectItem[][];
    containerHeight?: string;
}
export type TTypeOfWheel = "Time" | "Generic" | "Date";
