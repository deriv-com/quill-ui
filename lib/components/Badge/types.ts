import { TRegularSizes } from "@types";

export type TBadgeColorVariants = "success" | "warning" | "danger";
export type TBadgeVariant = "status" | "notification";

export type BadgeProps = {
    variant?: TBadgeVariant;
    size?: TRegularSizes;
    color?: TBadgeColorVariants;
    className?: string;
    label?: string;
    badgeClassName?: string;
};
