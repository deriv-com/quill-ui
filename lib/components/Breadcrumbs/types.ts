import { TMediumSizes } from "@types";
import { ReactNode } from "react";

export interface LinkProps {
    content: ReactNode;
    href: string;
    target?: "_blank" | "_self" | "_parent" | "_top";
}

export type BreadcrumbProps = {
    size?: TMediumSizes;
    links?: LinkProps[];
    className?: string;
};
