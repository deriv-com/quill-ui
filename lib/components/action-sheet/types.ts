import { ComponentPropsWithoutRef } from "react";

type OpenType =
    | {
          isOpen?: boolean;
          onOpen: () => void;
      }
    | {
          isOpen?: undefined;
          onOpen?: () => void;
      };

export type RootProps = ComponentPropsWithoutRef<"div"> &
    OpenType & {
        onClose?: () => void;
        type?: "modal" | "non-modal";
        expandable?: boolean;
    };

type ActionType = {
    content: string;
    onAction: () => void;
};

export type FooterProps = ComponentPropsWithoutRef<"div"> & {
    primaryAction?: ActionType;
    secondaryAction?: ActionType;
};

export type FooterAlignment = FooterProps;
