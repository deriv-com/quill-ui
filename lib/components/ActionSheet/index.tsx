import { ReactNode } from "react";
import ActionSheetRoot from "./root";
import ActionSheetHeader from "./header";
import ActionSheetContent from "./content";
import ActionSheetFooter from "./footer";
import ActionSheetHandleBar from "./handle-bar";
import ActionSheetTrigger from "./trigger";
import ActionSheetPortal from "./portal";

type ActionSheetType = {
    Root: typeof ActionSheetRoot;
    Header: typeof ActionSheetHeader;
    Content: typeof ActionSheetContent;
    Footer: typeof ActionSheetFooter;
    HandleBar: typeof ActionSheetHandleBar;
    Trigger: typeof ActionSheetTrigger;
    Portal: typeof ActionSheetPortal;
};

export const ActionSheet: ActionSheetType = ({
    children,
}: {
    children: ReactNode;
}) => {
    return <>{children}</>;
};

ActionSheet.Root = ActionSheetRoot;
ActionSheet.Header = ActionSheetHeader;
ActionSheet.Content = ActionSheetContent;
ActionSheet.Footer = ActionSheetFooter;
ActionSheet.HandleBar = ActionSheetHandleBar;
ActionSheet.Trigger = ActionSheetTrigger;
ActionSheet.Portal = ActionSheetPortal;

export default ActionSheet;
