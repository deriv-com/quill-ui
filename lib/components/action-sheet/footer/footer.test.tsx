import { screen, render } from 'test-utils'
import ActionSheet from '..'
import userEvent from '@testing-library/user-event'
import { FooterAlignment } from '../types'

describe('<ActionSheet.Footer/>', () => {
  it('should not render footer if no action passed', () => {
    render(<ActionSheet.Footer aria-label="Footer" />)
    const footerEl = screen.queryByLabelText('Footer')
    expect(footerEl).not.toBeInTheDocument()
  })
  it('should render correctly with className', () => {
    const onActionButton = jest.fn()
    render(
      <ActionSheet.Footer
        className="px-50"
        aria-label="Footer"
        primaryAction={{ content: 'Primary action', onAction: onActionButton }}
      />,
    )
    const footerEl = screen.getByLabelText('Footer')
    expect(footerEl).toHaveClass('px-50')
  })
  it('should render with primaryAction correctly', async () => {
    const onActionButton = jest.fn()
    render(
      <ActionSheet.Footer
        primaryAction={{ content: 'Primary action', onAction: onActionButton }}
      />,
    )
    const primaryBtn = screen.getByRole('button', { name: 'Primary action' })
    expect(primaryBtn).toBeInTheDocument()
    await userEvent.click(primaryBtn)
    expect(onActionButton).toHaveBeenCalled()
  })
  it('should render secondaryAction correctly', async () => {
    const onSecondaryAction = jest.fn()
    render(
      <ActionSheet.Footer
        secondaryAction={{
          content: 'Secondary action',
          onAction: onSecondaryAction,
        }}
      />,
    )
    const secondaryBtn = screen.getByRole('button', {
      name: 'Secondary action',
    })
    expect(secondaryBtn).toBeInTheDocument()
    await userEvent.click(secondaryBtn)
    expect(onSecondaryAction).toHaveBeenCalled()
  })
  const alignments: FooterAlignment[] = ['vertical', 'horizontal']

  alignments.forEach((alignment) => {
    it(`should render correctly with ${alignment}`, () => {
      render(
        <ActionSheet.Footer
          alignment={alignment}
          aria-label={`${alignment} footer`}
          primaryAction={{ content: 'action', onAction: () => null }}
        />,
      )
      const footer = screen.getByLabelText(`${alignment} footer`)
      expect(footer).toMatchSnapshot()
    })
  })
})
