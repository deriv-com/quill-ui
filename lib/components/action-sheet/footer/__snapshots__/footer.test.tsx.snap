// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionSheet.Footer/> should render correctly with horizontal 1`] = `
<div
  aria-label="horizontal footer"
  class="quill-action-sheet--footer quill-action-sheet--footer__variant--horizontal"
>
  <button>
    action
  </button>
</div>
`;

exports[`<ActionSheet.Footer/> should render correctly with vertical 1`] = `
<div
  aria-label="vertical footer"
  class="quill-action-sheet--footer quill-action-sheet--footer__variant--vertical"
>
  <button>
    action
  </button>
</div>
`;
